recon/src/recon.rs

```diff
@@ -429,23 +428,97 @@ where
     }
 }
 
+/// Represents reasons the store is unwilling to persist keys and values
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum InvalidItem<K: Key> {
+    /// The key or event data could not be parsed
+    InvalidFormat {
+        /// The key associated with the invalid data
+        key: K,
+    },
+    /// The data did not have a valid signature
+    InvalidSignature {
+        /// The key associated with the invalid data
+        key: K,
+    },
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// Represents items the store may be able to persist in the future
+pub struct PendingItem<K>
+where
+    K: Key,
+{
+    /// The key this item needs in order to be processed.
+    pub required_key: K,
```

Comment:
We may not want to compute/return this and simply return the items and let recon retry them in the next batch (related to comments on `pending_cache.rs`)

Reply:
Would that approach risk the conversation getting stuck in the worst case?  Sending the same events over and over that are all blocked waiting on some other event in another part of the range that it never gets to?

Reply:
in a well behaving scenario, we shouldn't recompute ranges.. when I tested one side being bad, we kept descending and got to the end of a conversation and were a bit confused, with one side hanging up and the other trying to send some more data but finding no one.. and then we started over. we need add better tests for it (including if both sides are just talking "correct" nonsense) because it's possible something worse happens like both sides not getting what they want indefinitely, which is not supposed to happen.